### PR TITLE
rbd-mirror: lock loss during sync should wait for in-flight copies

### PR DESCRIPTION
**Pull request by dillaman**
Signed-off-by: Jason Dillaman <dillaman@redhat.com>

### CODE REVIEW

File: src/tools/rbd_mirror/image_sync/ImageCopyRequest.cc [hunk 1]
```diff
@@ -186,22 +186,22 @@ void ImageCopyRequest<I>::send_next_object_copy() {
...
-  ++m_current_ops;
-
   Context *finish_op_ctx = nullptr;
   if (m_local_image_ctx->exclusive_lock != nullptr) {
     finish_op_ctx = m_local_image_ctx->exclusive_lock->start_op();
   }
   if (finish_op_ctx == nullptr) {
     derr << ": lost exclusive lock" << dendl;
-    finish(-EROFS);
+    if (m_ret_val == 0) {
```
Comment (trociny) [inline src/tools/rbd_mirror/image_sync/ImageCopyRequest.cc]: nit: the check for `m_ret_val == 0` looks like not necessary.

**Comment by dillaman** [comment]
@trociny tweaked it a little bit since I realized loss of the remote during a single object copy could deadlock the exclusive lock release.

**Comment by trociny** [comment]
@dillaman See jenkins failure (TestMockImageSyncImageCopyRequest.SimpleImage)

**Comment by dillaman** [comment]
@trociny update pushed

**Pull request merged**